### PR TITLE
feat(cli): Add ``--schema`` and ``--exclude`` option to route CLI.

### PR DESCRIPTION
**Pull request by andrewdoh**
### Pull Request Checklist
    

- [x]  New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [x] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR

 ### Description
 
 This PR adds --schema and --exclude option to litestar routes CLI command to include schema that are generated by openAPI and adds the ability to also exclude endpoints by passing multiple regex patterns.

per the discussions here: https://github.com/litestar-org/litestar/pull/1637. i've changed the functionality to default to just user defined routes. the --schema will add the openAPI generated ones if passed. Also, theres the exclude which can be passed multiple times and removes from the query anything that matches the regex passed.
 ### Close Issue(s)
 
Closes #1637

### CODE REVIEW

**Comment by andrewdoh** [comment]
@all-contributors please add @andrewdoh  for contributions

**Comment by andrewdoh** [comment]
@all-contributors please add @andrewdoh  for Code

File: tests/unit/test_cli/test_core_commands.py [hunk 1]
```diff
@@ -321,3 +324,98 @@ def test_run_command_with_server_lifespan_plugin(
         ssl_certfile=None,
         ssl_keyfile=None,
     )
+
+
+@pytest.mark.parametrize(
+    "app_content, schema_enabled, exclude_pattern_list",
+    [
+        (APP_FILE_CONTENT_ROUTES_EXAMPLE, False, ()),
+        (APP_FILE_CONTENT_ROUTES_EXAMPLE, False, ("/foo", "/destroy/.*", "/java", "/haskell")),
```
Comment (JacobCoffee) [inline tests/unit/test_cli/test_core_commands.py:333]: why taint the repo with java :(
/s

**Review by JacobCoffee — APPROVED** [review]

**Comment by JacobCoffee** [comment]
mostly lgtm once tests are passing

**Comment by provinzkraut** [comment]
@andrewdoh Please rebase from `develop` and change the target to `main`. As this is a new feature, it will go into the next minor release, which is being developed on `develop`.

**Comment by JacobCoffee** [comment]
@provinzkraut wonder if we can GH actions this with a keyword in a comment to trigger

**Comment by JacobCoffee** [comment]
@andrewdoh i think he meant `develop` there sorry.

**Comment by andrewdoh** [comment]
> @andrewdoh i think he meant `develop` there sorry.

no worries, thanks for handling it, I wasn't sure how to go about that.

**Comment by JacobCoffee** [comment]
@all-contributors please add @andrewdoh for test, docs

File: docs/usage/cli.rst [hunk 2]
```diff
@@ -185,6 +185,19 @@ The ``routes`` command displays a tree view of the routing table.
...
+| Flag            | Description                                                                                                                                           |
++=================+=======================================================================================================================================================+
+| ``--schema``    | Include default auto generated openAPI schema routes                                                                                                  |
++-----------------+ ------------------------------------------------------------------------------------------------------------------------------------------------------+
+| ``--exclude``   | Exclude endpoints from query with given regex patterns. multiple excludes allowed. e.g.  litestar routes  --schema --exclude=routes/.* --exclude=[]   |
++-----------------+-------------------------------------------------------------------------------------------------------------------------------------------------------+
```
Comment (JacobCoffee) [inline docs/usage/cli.rst:197]: Unsure why this isnt rendering properly

File: docs/usage/cli.rst [hunk 3]
```diff
@@ -185,6 +185,19 @@ The ``routes`` command displays a tree view of the routing table.
...
 
+Options
+~~~~~~~
+
++-----------------+-------------------------------------------------------------------------------------------------------------------------------------------------------+
+| Flag            | Description                                                                                                                                           |
++=================+=======================================================================================================================================================+
+| ``--schema``    | Include default auto generated openAPI schema routes                                                                                                  |
++-----------------+ ------------------------------------------------------------------------------------------------------------------------------------------------------+
+| ``--exclude``   | Exclude endpoints from query with given regex patterns. multiple excludes allowed. e.g.  litestar routes  --schema --exclude=routes/.* --exclude=[]   |
```
Comment (JacobCoffee) [inline docs/usage/cli.rst:196]:
```suggestion
| ``--exclude``   | Exclude endpoints from query with given regex patterns. Multiple excludes allowed. e.g., ``litestar routes  --schema --exclude=routes/.* --exclude=[]``   |
```

File: litestar/cli/_utils.py [hunk 4]
```diff
@@ -539,3 +542,34 @@ def _generate_self_signed_cert(certfile_path: Path, keyfile_path: Path, common_n
...
+    routes: list[HTTPRoute | ASGIRoute | WebSocketRoute], patterns: tuple[str, ...]
+) -> list[HTTPRoute | ASGIRoute | WebSocketRoute]:
+    regex_routes = []
+    valid_patterns = []
+    for pattern in patterns:
+        try:
+            check_pattern = re.compile(pattern)
+            valid_patterns.append(check_pattern)
+        except re.error as e:
+            console.print(f"Error: {e}. Invalid regex pattern supplied: {pattern}. omitting from querying results.")
```
Comment (JacobCoffee) [inline litestar/cli/_utils.py:557]:
```suggestion
            console.print(f"Error: {e}. Invalid regex pattern supplied: '{pattern}'. Omitting from querying results.")
```

File: docs/usage/cli.rst (hunk 2 above)
Comment (JacobCoffee) [reply docs/usage/cli.rst]: image

Comment (andrewdoh) [reply docs/usage/cli.rst]: i think it was missing a - and the | wasn't aligned with the other ones. it looks good now.
image

**Comment by JacobCoffee** [comment]
Sorry you had to fix that table, I was about to push my fix for my bad review and the baby started yelling

**Comment by andrewdoh** [comment]
> Sorry you had to fix that table, I was about to push my fix for my bad review and the baby started yelling

no worries it was my error haha you have much more important matters to attend to. thanks for pointing it out it lead to me the right path.

**Pull request merged**